### PR TITLE
Stack overview security

### PR DESCRIPTION
**Pull request by jyasveer**
1. Added the security tile card in overview section
2. Added the license card with the chart in overview section
3. Added the new columns to the components grid.
4. Style fixes

### CODE REVIEW

File: src/app/stack/overview/overview.component.html [hunk 1]
```diff
@@ -1,17 +1,49 @@
 <div class="overview-stack">
   <h1>{{messages && messages.title}}</h1>
-  <div class="row">
-    <!--<div class="col-sm-4 col-xs-12">
-      <div class="overview-cards card-pf card-pf-utilization">
+  <div *ngIf="stackOverviewData.noOfComponents === 0" class="alert alert-warning">
+    <span class="pficon pficon-warning-triangle-o"></span>
+    <strong>No Dependencies</strong> data available
```
Comment (arunkumars08) [inline src/app/stack/overview/overview.component.html]: Could you move this to constants?

File: src/app/stack/overview/overview.component.html [hunk 2]
```diff
@@ -1,17 +1,49 @@
...
           </h2>
         </div>
+        <div class="card-pf-body">
+          <div *ngIf="securityInfo && securityInfo.value === -1" class="row make-center">
+            <div class="pficon pficon-help warning-icon"></div>
+            <div class="cvss-id">No CVE or CVSS data available</div>
+          </div>
+          <div *ngIf="securityInfo && securityInfo.value >= 0 && securityInfo.value <= 6.9" class="row make-center">
+            <div class="fa fa-check-circle-o warning-icon"></div>
+            <div class="cvss-id">No security alerts</div>
```
Comment (arunkumars08) [inline src/app/stack/overview/overview.component.html]: Could you move this to constants?

File: src/app/stack/overview/overview.component.html [hunk 3]
```diff
@@ -61,24 +93,22 @@ <h2 class="card-pf-title ">
...
-              </div>
+        <div class="card-pf-body">
+          <div *ngIf="licenseChartInfo && licenseChartInfo.data.columns.length !== 0" class="row">
+            <div class="col-sm-12">
+              <f8-chart [data]="licenseChartInfo.data" [chartOptions]="licenseChartInfo.chartOptions" [configs]="licenseChartInfo.configs"></f8-chart>
             </div>
           </div>
+          <div *ngIf="licenseChartInfo && licenseChartInfo.data.columns.length === 0" class="row make-center">
+            <div class="pficon pficon-help warning-icon"></div>
+            <div class="cvss-id">No License data available</div>
```
Comment (arunkumars08) [inline src/app/stack/overview/overview.component.html]: Could you move this to constants?

File: src/app/stack/overview/overview.component.ts [hunk 4]
```diff
@@ -1,5 +1,7 @@
 import { Component, Input, OnInit, OnChanges } from '@angular/core';
+import * as _ from 'lodash';
```
Comment (arunkumars08) [inline src/app/stack/overview/overview.component.ts]: Why is this imported?

File: src/app/stack/stack-components/stack-components.component.ts [hunk 5]
```diff
@@ -177,25 +209,55 @@ export class StackComponents implements OnChanges {
...
+                    name: 'Avgerage Cyclomatic Complexity',
+                    identifier: this.keys['avgCycloComplexity']
+                }, {
+                    name: 'Total Files',
+                    identifier: this.keys['noOfFiles'],
                     isSortable: true
                 }
+                // , {
+                //     name: 'Public Popularity',
+                //     identifier: this.keys['publicPopularity']
```
Comment (invincibleJai) [inline src/app/stack/stack-components/stack-components.component.ts]: we can remove comments

File: src/app/stack/overview/overview.component.html (hunk 1 above)
Comment (jyasveer) [inline src/app/stack/overview/overview.component.html]: done

File: src/app/stack/overview/overview.component.html (hunk 2 above)
Comment (jyasveer) [inline src/app/stack/overview/overview.component.html]: done

File: src/app/stack/overview/overview.component.html (hunk 3 above)
Comment (jyasveer) [inline src/app/stack/overview/overview.component.html]: done

File: src/app/stack/overview/overview.component.ts (hunk 4 above)
Comment (jyasveer) [inline src/app/stack/overview/overview.component.ts]: i am using it below to do a groupBy operation

File: src/app/stack/stack-components/stack-components.component.ts (hunk 5 above)
Comment (jyasveer) [inline src/app/stack/stack-components/stack-components.component.ts]: removed

**Comment by invincibleJai** [comment]
@jyasveer push the changes

**Pull request merged**